fplus-lib/src/core/application/lifecycle.rs

```diff
@@ -116,13 +117,37 @@ impl LifeCycle {
         self.active_request
     }
 
-    pub fn reached_total_datacap(self) -> Self {
-        let empty = "".to_string();
-
+    pub fn close_application(self) -> Self {
         LifeCycle {
+            state: AppState::DeclineApplication,
```

Comment:
`Declined` specifically means that someone reviewed the application and rejected it on some grounds. Closing application is a more generic term, that could also apply in cases like client withdrawing the application or reaching the total requested datacap.
In fact, `close_application` is called when total datacap is reached - setting the state to `Declined` in such case is very confusing. It should be a separate state (or even just leave it as Granted, just mark as inactive?)